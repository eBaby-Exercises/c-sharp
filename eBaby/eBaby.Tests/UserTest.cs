using System;
using FluentAssertions;
using Microsoft.VisualBasic;
using Xunit;

namespace eBaby.Tests
{
    public class UserTest
    {
        [Fact]
        public void User_Creation()
        {
            string firstName = "James";
            string lastName = "Bond";
            string userEmail = "somebody@example.com";
            string userName = "[username]";
            string password = "[pWord]";

            var sut = new User(firstName, lastName, userEmail, userName, password);

            sut.FirstName.Should().Be(firstName);
            sut.LastName.Should().Be(lastName);
            sut.UserEmail.Should().Be(userEmail);
            sut.UserName.Should().Be(userName);
            sut.Password.Should().Be(password);
        }

        [Fact]
        public void User_Registration()
        {
            User user = Arbitrary.User();
            UserRegistry registry = new UserRegistry();
            registry.Add(user);
            registry.RegisteredUsers.Should().BeEquivalentTo(new[] {user});
        }

        [Fact]
        public void RetrievingUserByUserName_Should_ReturnCorrectUser()
        {
            User user = Arbitrary.User();
            User otherUser = Arbitrary.User();
            UserRegistry registry = new UserRegistry();
            registry.Add(user);
            registry.Add(otherUser);
            var foundUser = registry.FindUser(user.UserName);
            foundUser.Should().BeSameAs(user);
            foundUser = registry.FindUser(otherUser.UserName);
            foundUser.Should().BeSameAs(otherUser);
        }

        [Fact]
        public void User_Login_Success()
        {

            User user = Arbitrary.User();
            UserRegistry registry = new UserRegistry();
            registry.Add(user);
            var foundUser = registry.FindUser(user.UserName);
            foundUser.Should().BeSameAs(user);

            // Compare user's password with supplied password
            // If equal, then: set authenticated property to true
            // If not equal, then: throw BadCredentials exception
            // Verify that user logged in

        }
    }
}
