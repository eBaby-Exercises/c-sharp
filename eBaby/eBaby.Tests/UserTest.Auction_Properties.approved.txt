Auction { Seller = User { FirstName = , LastName = , UserEmail = , UserName = right_username, Password = right_password, IsLoggedIn = False, IsSeller = False }, Itemdescr = ItemDescr, Startprice = 23.95, Starttime = StartTime, Endtime = EndTime }